plugins/CorePluginsAdmin/Controller.php

```diff
@@ -87,6 +99,13 @@ public function __construct(Translator $translator, Plugin\SettingsProvider $set
 
     public function uploadPlugin()
     {
+        if (!$this->passwordVerify->isPasswordCorrect(
+            Piwik::getCurrentUserLogin(),
+            Common::getRequestVar('confirmPassword')
```

Comment:
probably doesn't make a difference but might be good to set `'confirmPassword', null, 'string'` to make sure we are getting a string and not something else (eg could be maybe an array not sure)

Reply:
OK, changed